code/game/machinery/computer/station_alert.dm

```diff
@@ -3,16 +3,45 @@
 	desc = "Used to access the station's automated alert system."
 	icon_screen = "alert:0"
 	icon_keyboard = "atmos_key"
-	circuit = /obj/item/circuitboard/computer/stationalert
+	circuit = /obj/item/circuitboard/computer/station_alert
 	light_color = LIGHT_COLOR_CYAN
 	/// Station alert datum for showing alerts UI
 	var/datum/station_alert/alert_control
 
+/obj/machinery/computer/station_alert/examine(mob/user)
+	. = ..()
+	var/obj/item/circuitboard/computer/station_alert/my_circuit = circuit
+	. += span_info("The console is set to [my_circuit.station_only ? "track all station and mining alarms" : "track alarms on the same z-level"].")
+
 /obj/machinery/computer/station_alert/Initialize(mapload)
-	alert_control = new(src, list(ALARM_ATMOS, ALARM_FIRE, ALARM_POWER), list(z), title = name)
+	// If the console's circuit is station_only, set up the alert_control to only listen to station areas
+	var/obj/item/circuitboard/computer/station_alert/my_circuit = circuit
+	if(my_circuit.station_only)
+		name = "station alert console"
+		var/list/alert_areas
+		alert_areas = (GLOB.the_station_areas + typesof(/area/mine))
+		alert_control = new(src, list(ALARM_ATMOS, ALARM_FIRE, ALARM_POWER), listener_areas = alert_areas, title = name)
+	else // If not station_only, just check the z-level
+		name = "local alert console"
+		alert_control = new(src, list(ALARM_ATMOS, ALARM_FIRE, ALARM_POWER), list(z), title = name)
 	RegisterSignals(alert_control.listener, list(COMSIG_ALARM_LISTENER_TRIGGERED, COMSIG_ALARM_LISTENER_CLEARED), PROC_REF(update_alarm_display))
```

Comment:
move this into a proc called like, link_alert() or something, have it `QDEL_NULL(alert_control)`, that will clean out our registered signals just fine